libmisc/getdate.y

```diff
@@ -395,7 +401,8 @@ relunit	: tUNUMBER tYEAR_UNIT {
 
 number	: tUNUMBER
           {
-	    if ((yyHaveTime != 0) && (yyHaveDate != 0) && (yyHaveRel == 0))
+	    if ((yyHaveTime != 0 || $1 >= 100) && !yyHaveYear
+		&& (yyHaveDate != 0) && (yyHaveRel == 0))
```

Comment:
I think this looks correct, but it's not exactly easy to parse.
What do you think about moving this baroque check into its own
helper function?

Reply:
I'd like to do that but my understanding of yacc is low and I fear it'll take me a lot of time to do it. Do you mind if we keep the change as it is?